### ISSUE TITLE
getinfo or listfunds should show estimate of total available to spend

### ISSUE THREAD

**Issue by rustyrussell** [ElementsProject/lightning#1878]
Someone asked, and currently you need to manually sum listfunds output, then subtract the fees you'd need.  That seems like a job for getinfo.

**Comment by fixone** [comment]
using ```jq``` getting all available funds (fees notwithstanding) is as simple as
 
```
~/lightning/cli/lightning-cli listfunds | jq '.outputs | .[] | .value' | jq -s add
```
to get funds available in the c-lightning wallet as unspent outputs and
```
~/lightning/cli/lightning-cli listpeers | jq -r '.peers | to_entries[] | select (.value.channels[].state=="CHANNELD_NORMAL") | .value.channels[].msatoshi_to_us  ' | jq -s add | expr $(</dev/stdin) / 1000 | echo $(</dev/stdin)  "satoshi available"
```
to get all the funds available on our end in all NORMAL channels

**Comment by ZmnSCPxj** [comment]
Summary plugin in lightningd/plugins does this (`avail_out`), and more: https://github.com/lightningd/plugins/tree/master/summary

Should we consider "bless" summary?

**Comment by darosior** [comment]
Most people running C-lightning are, I believe, using summary for this purpose now. Do we still need a new field ?

**Comment by ZmnSCPxj** [comment]
If the feature is so useful, it might actually be better to put in the built-in plugins a C version.

**Comment by darosior** [comment]
Or directly pull  `summary.py` ?
/me hides

**Comment by ZmnSCPxj** [comment]
Best not to depend on python code for our "stock" release.  Because of the disasters in upgrading to Python 3.x, you really need the latest Python (3.7 or 3.8 or higher, there are some weird incompatibilities and outright bugs in lower Python 3.x series) to work on most Python code.  Unfortunately existing long-term-support OSs still come with Python 3.5 or lower, meaning to decently run the latest Python 3.x code you need to install pyenv.  This is an additional requirement that current C-Lightning does not actually have in order to *run* (it *is* a requirement for it to get *tested*, but users do not test the software they use anyway, they just run it).

**Comment by darosior** [comment]
Yeah you previously made a point regarding depending and i agree with you.

This is currently a build requirement and we get a lot of users coming to irc struggling with mako or mrkd.
-------- Original Message --------

**Comment by rustyrussell** [comment]
Generally, adding to getinfo is not the greatest idea: specific commands exist and we can't cover every possible useful thing!
